hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -70,6 +80,13 @@ class RobotHW : public InterfaceManager
 
   /** \brief The init function is called to initialize the RobotHW from a
    * non-realtime thread.
+   * 
+   * The registration of data members to interfaces is done via 
+   * \ref JointStateInterface, for read-only joints, or \ref JointCommandInterface, 
+   * for joints that accept commands and provide feedback (read and write), 
+   * including its derived classes (e.g. \ref VelocityJointInterface).
+   * These interfaces make use of \ref JointHandle or \ref JointStateHandle through 
+   * the \ref HardwareResourceManager.
```

Comment:
Again, I am not sure if the registration of interfaces belongs to init or the constructor.

Reply:
I think this should be reworded to be more generic - We are not limited to classes like `JointStateInterface`, `JointCommandInterface`, `JointHandle`, etc. Though these are the ones `ros_control` itself provides, users are completely free to define their own custom interfaces and handles.

Reply:
Good point, I didn't think about custom ones. I'll try to find a more generic description.